main.tf

```diff
@@ -50,7 +50,7 @@ resource "azurerm_dns_a_record" "records_a" {
 ## Below resource will add cname_record in DNS zone.
 ##-----------------------------------------------------------------------------
 resource "azurerm_dns_cname_record" "records_cname" {
-  for_each            = var.enabled ? { for record in var.cname_records : record.name => record } : {}
+  for_each            = var.enabled && var.enabled_dns ? { for record in var.cname_records : record.name => record } : {}
```

Comment:
what is the purpose to add the two control variables.  
like enabled and var.enabled_dns

Reply:
one is for public dns and other is for controlling complete module creation.

Reply:
Let me remove `enabled` one.